pkg/lib/bundle/validate.go

```diff
@@ -28,12 +58,20 @@ func (i imageValidator) PullBundleImage(imageTag, directory string) error {
 }
 
 // ValidateBundle takes a directory containing the contents of a bundle and validates
-// the format of that bundle for correctness
-func (i imageValidator) ValidateBundle(directory string) error {
+// the format of that bundle for correctness using these criteria
+// 1. Validate if the directory has two required directories for /manifests and /metadata
+// 2. Expecting bundle manifests files to be in /manifests and metadata files (including
+// annotations.yaml) to be in /metadata
+// 3. Validate the information in annotations to match the bundle contents such as
+// its media type, and channel information.
+// Inputs:
+// directory: the directory which the /manifests and /metadata exist
+// Outputs:
+// error: ValidattionError which contains a list of errors
+func (i imageValidator) ValidateBundleFormat(directory string) error {
 	var manifestsFound, metadataFound bool
 	var annotationsDir, manifestsDir string
-	var annotationErrors []error
-	var formatErrors []error
+	var validationErrors []error
 
 	items, _ := ioutil.ReadDir(directory)
```

Comment:
should we really drop the error here? there could be a permissions issue or something else going on when accessing the directory, and none of the subsequent validation would make any sense. Same for L225 with the content validation.

Reply:
We can certainly acknowledge the error here. The reason I keep I ignore the error is because we inspect the directory that is actually created by this problem. Users supply the image tag and the program will pull the image down and unpack the contents into directories that it creates. So there shouldn't be a permission issue here. This function is meant to be use in conjunction with other functions to create a validation flow. However, I understand from standalone API standpoint, maybe we should return every error that occurs to avoid confusions.